src/types.jl

```diff
@@ -101,6 +101,16 @@ struct ProjectedGradientAscent{T}
     stepsize::T
 end
 
+"""
+    ProjectedVariance{S,T}
+Project variance such that for whatever method used,
+minimum variance does not go below varfloor
```

Comment:
Minor tweaks:
```suggestion
    ProjectedVariance{S,T}

Update variance using `method` then project the variance
so that the variance iterate does not go below `varfloor`.
```